### ISSUE TITLE
After force deleting a deploy-as-is template, the VMs using it cannot start 

### ISSUE THREAD

**Issue by vladimirpetrov** [apache/cloudstack#7970]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
API, UI
~~~

##### CLOUDSTACK VERSION

~~~
4.16, 4.17, 4.18.1
~~~

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

N/A

##### SUMMARY

If a deploy-as-is template (VMWare vApp, for example) is removed using 'Force delete', the VMs using it won't be able to start properly because the details in the 'template_deploy_as_is_details' are removed.

##### STEPS TO REPRODUCE

~~~
Register a deploy-as-is template (enabling the Read VM settings from OVA ), start a VM with this template, then remove the template using 'Force delete' and stop/start the VM.

Note: There is another regression bug in 4.17 - 4.18.1 that makes reproducing this one much harder:
https://github.com/apache/cloudstack/issues/7998

So to reproduce it you'll need a 4.16 environment or 4.18 branch that includes the fix.
~~~

##### EXPECTED RESULTS

~~~
The VM starts normally.
~~~

##### ACTUAL RESULTS

~~~
The VM cannot start.
~~~

**Comment by DaanHoogland** [comment]
Vladi, I think if you use force delete you have the intention and thought about it. What would you expect to happen? I think for the VM to just keep working is not what the force delete functionality is for.

**Comment by weizhouapache** [comment]
> Vladi, I think if you use force delete you have the intention and thought about it. What would you expect to happen? I think for the VM to just keep working is not what the force delete functionality is for.

I think the deploy-as-it details of template should be kept.

**Comment by harikrishna-patnala** [comment]
@weizhouapache the problem is with this line
String value = property.isPassword() ? DBEncryptionUtil.decrypt(detail.getValue()) : detail.getValue();

where we are checking if the property is a password.

I'm thinking to add a new column something like "is_password" to the table "user_vm_deploy_as_is_details", this will remove the dependency on the template.
As an alternative as you said we can keep the template deploy as is details but thinking if it will keep those entries always in the database.

What you say ?

**Comment by harikrishna-patnala** [comment]
had a discussion with @weizhouapache, going with the solution to keep the template details after it is deleted.

**Comment by JoaoJandre** [comment]
Closing as this was fixed by #8115
